pkg/lib/oauth/handler/handler_authz.go

```diff
@@ -213,7 +242,8 @@ func (h *AuthorizationHandler) doHandleConsent(req *http.Request, withUserConsen
 			RedirectURI: nil,
 		}
 		if errors.As(err, &oauthError) {
-			resultErr.Response = oauthError.Response
+			resultErr := h.prepareErrInvalidOAuthResponse(req, *oauthError)
+			return resultErr, nil
```

Comment:
Convert this pattern

```
resultErr := authorizationResultError{
  RedirectURI: nil
}
resultErr := h.prepareErrInvalidOAuthResponse(req, *oauthError)
return resultErr, nil
```

To

```
resultErr := h.makeOAuthErrorResponse(req)
```

Rationale
1. Shadowing a local variable and return it in a if block is hard to read.
2. The original code does not have multiple returns. And after this change, sometimes it early returns, sometimes it does not. This is even more hard to read.